### PR TITLE
[#9420] Consolidate Recycle Bin RESTFul API

### PR DESCRIPTION
**Pull request by rrtheonlyone**
Part of #9420

**Changes**:

| New endpoint        | Method           | Old Endpoint in v6  |
| ------------- |:-------------:| :-----:|
| /bin/course   | DELETE | INSTRUCTOR_COURSES_RESTORE (PUT) |

Simple API rename for recycle bin API.

### CODE REVIEW

File: src/main/java/teammates/ui/webapi/action/ActionFactory.java [hunk 1]
```diff
@@ -83,7 +83,7 @@
         map(ResourceURIs.COURSE_ENROLL_STUDENTS, GET, GetCourseEnrollStudentsAction.class);
         map(ResourceURIs.INSTRUCTOR_COURSES, GET, GetInstructorCoursesAction.class);
         map(ResourceURIs.INSTRUCTOR_COURSES, POST, AddInstructorCourseAction.class);
-        map(ResourceURIs.INSTRUCTOR_COURSES_RESTORE, PUT, RestoreInstructorSoftDeletedCourseAction.class);
+        map(ResourceURIs.BIN_COURSE, DELETE, RestoreInstructorSoftDeletedCourseAction.class);
```
Comment (xpdavid) [inline src/main/java/teammates/ui/webapi/action/ActionFactory.java]: - You should rename `RestoreInstructorSoftDeletedCourseAction` to `RestoreCourseAction`.
- Put the mapping around the `course` related mappings to make it more organised.

File: src/test/java/teammates/test/cases/webapi/RestoreInstructorSoftDeletedCourseActionTest.java [hunk 2]
```diff
@@ -18,12 +18,12 @@
...
     @Override
     protected String getActionUri() {
-        return Const.ResourceURIs.INSTRUCTOR_COURSES_RESTORE;
+        return Const.ResourceURIs.BIN_COURSE;
     }
 
     @Override
     protected String getRequestMethod() {
-        return PUT;
+        return DELETE;
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/RestoreInstructorSoftDeletedCourseActionTest.java]: Take a look at the submissionParams below, I don't think you need `INSTRUCTOR_ID`?

Comment (rrtheonlyone) [reply src/test/java/teammates/test/cases/webapi/RestoreInstructorSoftDeletedCourseActionTest.java]: Yup your right! Removed that

File: src/main/java/teammates/ui/webapi/action/ActionFactory.java (hunk 1 above)
Comment (rrtheonlyone) [reply src/main/java/teammates/ui/webapi/action/ActionFactory.java]: Done!

File: src/main/java/teammates/ui/webapi/action/RestoreCourseAction.java [hunk 3]
```diff
@@ -9,7 +9,7 @@
 /**
  * Action: Restores a soft-deleted course from Recycle Bin.
```
Comment (xpdavid) [inline src/main/java/teammates/ui/webapi/action/RestoreCourseAction.java]: `Restore a course from Recycle Bin.`

File: src/test/java/teammates/test/cases/webapi/RestoreCourseActionTest.java [hunk 4]
```diff
@@ -37,15 +37,14 @@ public void testExecute() throws Exception {
         ______TS("Typical case, restore a soft-deleted course from Recycle Bin");
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/RestoreCourseActionTest.java]: You should add more tests here?

- Not enough parameter?
- Non-existent?
- Not in recycle bin?

Comment (rrtheonlyone) [reply src/test/java/teammates/test/cases/webapi/RestoreCourseActionTest.java]: Okay added 👍

File: src/main/java/teammates/ui/webapi/action/RestoreCourseAction.java (hunk 3 above)
Comment (rrtheonlyone) [reply src/main/java/teammates/ui/webapi/action/RestoreCourseAction.java]: Done

File: src/test/java/teammates/test/cases/webapi/RestoreCourseActionTest.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        String[] submissionParams = new String[] {
+                Const.ParamsNames.COURSE_ID, courseId,
+        };
+
+        RestoreCourseAction action = getAction(submissionParams);
+        JsonResult result = getJsonResult(action);
+        MessageOutput message = (MessageOutput) result.getOutput();
+
+        assertEquals(HttpStatus.SC_OK, result.getStatusCode());
+        assertEquals("The course " + courseId + " has been restored.", message.getMessage());
```
Comment (xpdavid) [inline src/test/java/teammates/test/cases/webapi/RestoreCourseActionTest.java]: You should fetch the course and validate also? Same for other places?

Comment (rrtheonlyone) [reply src/test/java/teammates/test/cases/webapi/RestoreCourseActionTest.java]: Will adding this `assertNull(CoursesLogic.inst().getSoftDeletedCourseForInstructor(instructor1OfCourse1));` be sufficient?

Comment (ChooJeremy) [reply src/test/java/teammates/test/cases/webapi/RestoreCourseActionTest.java]: 1. You can convert `CoursesLogic.inst().` -> `logic.` for most methods
2. Try `AssertNull(logic.getCourse(instructor1OfCourse1.courseId).deletedAt);`

Comment (rrtheonlyone) [reply src/test/java/teammates/test/cases/webapi/RestoreCourseActionTest.java]: yeah thats works! Thanks @ChooJeremy, have updated the tests accordingly

**Pull request merged**